boac-5130 - Profile View Complete

mohamalnadi opened the pull request:
Jira: https://jira-secure.berkeley.edu/browse/BOAC-5130

mohamalnadi committed:
boac-5130 - Profile View Complete

pauline2k commented on src/components/admin/DemoModeToggle.vue:
```diff
@@ -1,3 +1,5 @@
+Demo
```
@mohamalnadi Are these title rows outside the template a new convention for Vue 3 files, or were they meant to be temporary?

pauline2k commented on src/components/admin/DemoModeToggle.vue:
```diff
@@ -10,23 +12,26 @@
           >
-            <span class="sr-only">Demo mode is </span>{{ inDemoMode ? 'On' : 'Off' }}
-          </b-form-checkbox>
+            <!-- <span class="sr-only">Demo mode is </span>{{ inDemoMode ? 'On' : 'Off' }} -->
```
Commented-out screenreader text: intentional?

lyttam replied:
I think we want either a `<label>` element separate from the `<v-checkbox>` _or_ `label` and `aria-label` properties on the `<v-checkbox>`. In the past I've noticed that using both a separate label and label properties causes the screen reader to voice it twice.

If you go with the second option, the `aria-label` property can be set to `Demo mode is ${inDemoMode ? 'On' : 'Off'}`.

pauline2k commented:
Thanks @mohamalnadi! Looks good to me, a couple questions on specific lines. (Maybe @lyttam can take a look too since she's been closer to this work.)

mohamalnadi commented:
@pauline2k @lyttam - Made the new changes! Thank you! 

I am not sure where the word "Demo" came from outside of the template, I may have added it accidentally?

For the Demo On/Off label, I added an aria-label. The reason I removed the span is because in Vuetify, the nested span text aligned to the left of the checkbox, but you would have to use a label to have the text on the right of the checkbox.

lyttam commented:
Looks good, thanks!
